file/writable_file_writer.cc

```diff
@@ -490,19 +500,21 @@ IOStatus WritableFileWriter::WriteBuffered(
   size_t left = size;
   DataVerificationInfo v_info;
   char checksum_buf[sizeof(uint32_t)];
+  IOOptions io_options;
+  Env::IOPriority rate_limiter_priority_used =
+      WritableFileWriter::DecideRateLimiterPriority(
+          writable_file_->GetIOPriority(), op_rate_limiter_priority);
 
   while (left > 0) {
-    size_t allowed;
-    Env::IOPriority rate_limiter_priority_used =
-        WritableFileWriter::DecideRateLimiterPriority(
-            writable_file_->GetIOPriority(), op_rate_limiter_priority);
-    if (rate_limiter_ != nullptr &&
-        rate_limiter_priority_used != Env::IO_TOTAL) {
-      allowed = rate_limiter_->RequestToken(left, 0 /* alignment */,
-                                            rate_limiter_priority_used, stats_,
-                                            RateLimiter::OpType::kWrite);
-    } else {
-      allowed = left;
+    size_t allowed = left;
+    if (rate_limiter_priority_used != Env::IO_TOTAL) {
+      if (rate_limiter_ != nullptr) {
```

Comment:
I think we decided to always pass priority in IOOptions. What the file system does with it should be a contract between the user and the fs.

Reply:
@ajkr Please be aware about this final decision.

Reply:
I will update the logic.